server/src/internalClusterTest/java/org/opensearch/cluster/allocation/AwarenessAllocationIT.java

```diff
@@ -364,18 +364,22 @@ public void testThreeZoneOneReplicaWithForceZoneValueAndLoadAwareness() throws E
             .put("cluster.routing.allocation.awareness.force.zone.values", "a,b,c")
             .put("cluster.routing.allocation.load_awareness.skew_factor", "0.0")
             .put("cluster.routing.allocation.load_awareness.provisioned_capacity", Integer.toString(nodeCountPerAZ * 3))
+            .put("cluster.routing.allocation.allow_rebalance", "indices_primaries_active")
```

Comment:
@imRishN  From test failure trace (MasterNotDiscoveredException), it is not clear if it is an actual issue or a flaky one. Can you explain how existing test is identified as flaky and changes here fixes it ?

Reply:
The test here now adds a dedicated cluster manager node where as previously there was no dedicated cluster manager setup and the test was randomly killing half the nodes in a particular zone. I assume `MasterNotDiscoveredException` was coming when a node that was stopped was an active master that time and hence the exception was thrown sometimes.